sdk/src/kalix.ts

```diff
@@ -605,22 +607,54 @@ export class Kalix {
     process.exit(0);
   }
 
+  /** @internal */
+  reportError(error: discovery.UserFunctionError): void {
+    const msg = this.reportErrorLogic(
+      error.code,
+      error.message,
+      error.detail,
+      error.sourceLocations,
+      error.severity,
+    );
+    switch (error.severity as number) {
+      case UserFunctionErrorSeverity.INFO:
+      case UserFunctionErrorSeverity.WARNING:
+        console.log(msg);
+        break;
+      default:
+        console.error(msg);
+    }
+  }
+
   /** @internal */
   reportErrorLogic(
-    code?: string,
-    message?: string,
-    detail?: string,
-    locations?: Array<discovery.SourceLocation>,
-  ) {
-    let msg = `Error reported from Kalix system: ${code} ${message}`;
+    code: string,
+    message: string,
+    detail: string,
+    locations: Array<discovery.SourceLocation>,
+    severity: discovery.Severity,
+  ): string {
+    let messageType = 'Error';
+
+    switch (severity as number) {
+      case UserFunctionErrorSeverity.INFO:
+        messageType = 'Message';
+        break;
+      case UserFunctionErrorSeverity.WARNING:
+        messageType = 'Warning';
+        break;
+    }
+
+    let msg = `${messageType} reported from Kalix system: ${code} ${message}`;
```

Comment:
BTW, shouldn't this be JSON anyway?

Reply:
Right, no structured JSON logging anywhere at the moment. Older issue is https://github.com/lightbend/kalix-proxy/issues/618. Could make sense to have our own small logging API, that users can also make use of, from the context or something.